### PR TITLE
Hw fruit shop tests

### PR DESCRIPTION
**Pull request by OlehMytiai**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+public class FileReaderServiceCsvTest {
+    private static final String FILE_NAME_OK = "src/test/resources/database_ok.csv";
+    private static final String EMPTY_FILE_NAME = "src/test/resources/database_empty.csv";
+    private FileReaderService fileReaderService;
+
+    @Before
+    public void setUp() {
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java:17]: you are creating `new fileReader` after each test, use `@BeforeClass` instead, do that in all other classes

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        List<String> actual = fileReaderService.readFromFile(FILE_NAME_OK);
+
+        assertEquals(8, actual.size());
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readData_fileNotExist() {
+        List<String> expected = new ArrayList<>();
+        assertEquals(expected, fileReaderService.readFromFile(""));
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java:42]: I think we dont even need `AssertEquals` here
```suggestion
       fileReaderService.readFromFile("");
```

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public void readData_emptyFile() {
+        List<String> expected = new ArrayList<>();
+        List<String> actual = fileReaderService.readFromFile(EMPTY_FILE_NAME);
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readData_null_notOk() {
+        List<String> expected = new ArrayList<>();
+        assertEquals(expected, fileReaderService.readFromFile(null));
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java:55]: same here
```suggestion
        fileReaderService.readFromFile(null);
```

**Review by karma-o — APPROVED** [review]

File: src/test/java/core/basesyntax/service/FruitTransactionTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.FruitTransaction.Operation;
+import java.util.NoSuchElementException;
+import org.junit.Test;
+
+public class FruitTransactionTest {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/FruitTransactionTest.java:9]: There is no need to test dto objects

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readData_fileNotExist() {
+        fileReaderService.readFromFile("");
+    }
+
+    @Test
+    public void readData_emptyFile() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java:45]:
```suggestion
    public void readData_emptyFile_ok() {
```

File: src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        expected.add("s,banana,50");
+
+        List<String> actual = fileReaderService.readFromFile(FILE_NAME_OK);
+
+        assertEquals(8, actual.size());
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readData_fileNotExist() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FileReaderServiceCsvTest.java:40]:
```suggestion
    public void readData_fileNotExist_notOk() {
```

File: src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+        String expected = "fruit,quantity"
+                + System.lineSeparator() + "banana,152"
+                + System.lineSeparator() + "apple,90";
+        String actual = fruitShopService.makeDailyReport(data);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void makeDailyReport_emptyList() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java:51]:
```suggestion
    public void makeDailyReport_emptyList_ok() {
```

File: src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        List<String> data = new ArrayList<>();
+        String expected = "fruit,quantity";
+        String actual = fruitShopService.makeDailyReport(data);
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void makeDailyReport_null() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java:59]:
```suggestion
    public void makeDailyReport_null_notOk() {
```

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/impl/FruitReportCreateServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+        String expected = "fruit,quantity"
+                + System.lineSeparator() + "banana,199"
+                + System.lineSeparator() + "apple,140";
+        String actual = reportCreateService.createReport();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void createReport_emptyStorage() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FruitReportCreateServiceTest.java:41]:
```suggestion
    public void createReport_emptyStorage_ok() {
```

**Review by KostyaNikiforov — APPROVED** [review]

**Pull request closed without merging**